controllers/propagator/replicatedpolicy_controller.go

```diff
@@ -219,45 +227,40 @@ func (r *ReplicatedPolicyReconciler) Reconcile(ctx context.Context, request ctrl
 		return reconcile.Result{}, err
 	}
 
-	instanceGVK := desiredReplicatedPolicy.GroupVersionKind()
-	instanceObjID := k8sdepwatches.ObjectIdentifier{
-		Group:     instanceGVK.Group,
-		Version:   instanceGVK.Version,
-		Kind:      instanceGVK.Kind,
-		Namespace: request.Namespace,
-		Name:      request.Name,
-	}
-
 	// save the watcherError for later, so that the policy can still be updated now.
 	var watcherErr error
 
-	if policyHasTemplates(rootPolicy) {
-		if replicatedExists {
-			// If the replicated policy has an initialization vector specified, set it for processing
-			if initializationVector, ok := replicatedPolicy.Annotations[IVAnnotation]; ok {
-				tempAnnotations := desiredReplicatedPolicy.GetAnnotations()
-				if tempAnnotations == nil {
-					tempAnnotations = make(map[string]string)
-				}
+	if replicatedExists {
+		// If the replicated policy has an initialization vector specified, set it for processing
+		if initializationVector, ok := replicatedPolicy.Annotations[IVAnnotation]; ok {
+			tempAnnotations := desiredReplicatedPolicy.GetAnnotations()
+			if tempAnnotations == nil {
+				tempAnnotations = make(map[string]string)
+			}
 
-				tempAnnotations[IVAnnotation] = initializationVector
+			tempAnnotations[IVAnnotation] = initializationVector
 
-				desiredReplicatedPolicy.SetAnnotations(tempAnnotations)
-			}
+			desiredReplicatedPolicy.SetAnnotations(tempAnnotations)
 		}
+	}
 
-		// Any errors to expose to the user are logged and recorded in the processTemplates method. Only retry
-		// the request if it's determined to be a retryable error (i.e. don't retry syntax errors).
-		err := r.processTemplates(ctx, desiredReplicatedPolicy, decision.Cluster, rootPolicy)
-		if errors.Is(err, ErrRetryable) {
-			// Return the error if it's retryable, which will utilize controller-runtime's exponential backoff.
-			return reconcile.Result{}, err
-		}
-	} else {
-		watcherErr := r.TemplateResolver.UncacheWatcher(instanceObjID)
-		if watcherErr != nil {
-			log.Error(watcherErr, "Failed to uncache objects related to the replicated policy's templates")
-		}
+	// Any errors to expose to the user are logged and recorded in the processTemplates method. Only retry
+	// the request if it's determined to be a retryable error (i.e. don't retry syntax errors).
+	tmplErr := r.processTemplates(ctx, desiredReplicatedPolicy, decision.Cluster, rootPolicy)
+	if errors.Is(tmplErr, ErrRetryable) {
+		// Return the error if it's retryable, which will utilize controller-runtime's exponential backoff.
+		return reconcile.Result{}, tmplErr
+	} else if errors.Is(tmplErr, ErrSAMissing) {
```

Comment:
`else` is not needed?

Reply:
It's not because template errors that are not retryable must continue so that the replicated policy has the hub template error annotation set.